packages/data/src/plugins/actions.js

```diff
@@ -125,6 +126,62 @@ export function* installAndActivatePlugins( plugins ) {
 	}
 }
 
+export const createErrorNotice = ( errorMessage ) => {
+	return dispatch( 'core/notices', 'createNotice', errorMessage );
+};
+
+export function* installJetpackAndConnect() {
+	try {
+		yield dispatch( STORE_NAME, 'installPlugins', [ 'jetpack' ] );
+		yield dispatch( STORE_NAME, 'activatePlugins', [ 'jetpack' ] );
+
+		const url = yield select( STORE_NAME, 'getJetpackConnectUrl', {
+			redirect_url: getAdminLink( 'admin.php?page=wc-admin' ),
+		} );
+
+		// getJetpackConnectUrl doesn't throw errors, it sets error state if
+		// something went wrong.
+		const error = yield select(
+			STORE_NAME,
+			'getPluginsError',
+			'getJetpackConnectUrl'
+		);
+
+		if ( error ) {
+			yield createErrorNotice( error );
+		} else {
+			window.location = url;
+		}
+	} catch ( error ) {
+		yield createErrorNotice( error.message );
```

Comment:
I feel the data package should remain data centric and avoid interactions with other data stores when possible as well as avoid any opinionated UI changes.

The consumer of these data stores may not want to show notices depending on their use case, but I do see the benefit and DRYness of creating notices directly from the store.  Up for debate, but whichever we choose I think we should be consistent across data stores.

Reply:
@joshuatf I def had a double take when writing this code too, I think I could probably adjust this so we can pass a error callback of some kind instead, letting people do whatever they want when the error happens. I'll explore this.

Reply:
If we throw an error inside this catch block, the consumer can then catch and decide what to do with it.  I like this along with setting the error in the data store so that components have access to the data on both sides.

https://github.com/woocommerce/woocommerce-admin/blob/9be744b63a3beb56b5d94abf8a0725b8822de722/packages/data/src/plugins/actions.js#L83

>One reason I thought this was ok was that it deals with a core store, but then... maybe not every implementer will want to use core/notices? Anyway I'm gonna investigate making this more agnostic.

I think you're right in that most of the time, usage of core/notices will not be a problem, but I do like the separation of this concern and making the data more agnostic.

Did you want to tackle here or a follow-up?

Reply:
@joshuatf I'm having a go at solving this one here. Thanks, the throw is a good suggestion